File: java/data-structure/queue/src/main/java/ru/reimu/queue/IQueue.java
package ru.reimu.queue;

/**
 * @Author: Tomonori
 * @Date: 2020/2/17 15:09
 * @Title:
 * @Desc: ↓ ↓ ↓ ↓ ↓
 * -----
 */
public interface IQueue<T> {

    int getSize();

    boolean isEmpty();

    void enqueue(T t);

    T dequeue();

    T getFront();
}

File: golang/data-structure/queue/IQueue.go
package queue

/**
 * @Author: Tomonori
 * @Date: 2020/2/18 14:51
 * @Title:
 * --- --- ---
 * @Desc:
 */
type IQueue interface {
	GetSize() int

	IsEmpty() bool

	Enqueue(t interface{}) error

	Dequeue() (interface{}, error)

	GetFront() (interface{}, error)
}
